lib/common/socket.c

```diff
@@ -999,6 +1039,37 @@ void h2o_socket_ssl_async_resumption_setup_ctx(SSL_CTX *ctx)
     /* if necessary, it is the responsibility of the caller to disable the internal cache */
 }
 
+static void on_dispose_ssl_ctx_ex_data(void *parent, void *ptr, CRYPTO_EX_DATA *ad, int idx, long argl, void *argp)
+{
+    h2o_cache_t *ssl_session_cache = (h2o_cache_t *)ptr;
+    if (ssl_session_cache != NULL)
+        h2o_cache_destroy(ssl_session_cache);
+}
+
+static int get_ssl_session_cache_index()
+{
+    static int index = INT_MAX;
```

Comment:
Initial value should be set to a value that is never returned by `SSL_CTX_get_ex_new_index`. Please consult the manual of the function to determine the appropriate value.